[CS2113-T14-3] MyLedger

dsicol opened the pull request:
This product is for university students who prefer CLI over GUI to manage their finances while in University, as this is when student income is minimal.

KedrianLoh commented on docs/team/images/saveList.png:
Alignment of the saveList() and saveExpenditureList to the start of the activation bar can be improved.

KedrianLoh commented on docs/team/images/initializeList.png:
Control of the diagram becomes ambiguous after addExpenditure(). The activation bars are mostly too long and not properly aligned.

KedrianLoh commented on docs/team/images/parserEdit.png:
When the method is returned, the line should be dotted instead of solid. Alignment of the activation bars can be further improved.

thomasjlalba commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,342 @@
+The following shows the UML diagram used for the parser component implemented in MyLedger.
+
+<p align="center">
+    <img src="team/images/parserOverview.png">
```
At the bottom of the picture, there would need to be a closure in the activation bar of :MainInputParser, it seems like currently there is no closure

thomasjlalba replied:
The picture in the Github Page also seems to be cut out (on the right side).

jaredoong replied:
Image is cut off on the right side. Additionally, the figure is a little small due to the multiple options shown. One possible workaround could be just to include only 2 options just to show an example, while clearly stating that the other options are omitted from the diagram to prevent cluttering it.

thomasjlalba commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,342 @@
+updated with all the current expenditures in the expenditure array list.
+
+<p align="center">
+    <img src="team/images/saveList.png">
```
I don't think the method call does not need to be underlined

thomasjlalba commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,342 @@
+The sequence diagram below shows the interactions of a successful execution of the EditCommand
+
+<p align="center">
+    <img src="team/images/parserEdit.png">
```
The termination (X) should be after the activation bar (by a dotted line), I don't think it should be terminated right at the end of activation bar.

jaredoong replied:
Activation bar for `getExpenditure(index)` should start at the head of arrow instead of above it. The 'X' that represents termination could also possibly be shifted down to below the activation bar.

thomasjlalba reviewed (commented):
I think the Developer Guide is quite in depth and seems to explain quite well. Just some minor issues here and there in terms of the diagrams. I think if it is possible can reformat the pictures as there are some pictures that are very long and so its very small on the Github page. 🥇

yixuann02 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,342 @@
+
+The following shows the UML diagram used for the Expenditure Categories component implemented in MyLedger.
+
+<p align="center">
+    <img src="team/images/UMLClassDiagramExpenditure.png" width="80%">
+    <br/>
+    <i>Figure 3: UML diagram for the Expenditure Categories component</i>
+</p>
+
```
Very clear and easy to read UML Diagram provided. Also, good explanation of the diagram.

yixuann02 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,342 @@
+
+The `ViewDateExpenditureCommand` and `ViewTypeExpenditureCommand` classes contain the operations pertaining to displaying a filtered expenditure list by the expenditure date and type respectively.
+
+Below shows the UML diagram representing the `command` package.
+<p align="center">
+    <img src="team/images/umlCommandClassDiagram.png" width="100%">
+    <br/>
+    <i>Figure 4: UML diagram for the command package</i>
+</p>
+
```
Perhaps you could split the UML diagram into different parts.  The UML diagram is too small and unreadable.

yixuann02 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,342 @@
+After every user input is completed, saveExpenditureList is called, and the text file will be
+updated with all the current expenditures in the expenditure array list.
+
+<p align="center">
+    <img src="team/images/saveList.png">
+    <br/>
+    <i>Figure 5: Sequence diagram for TxtFileStatus</i>
+</p>
```
The activation bar of a method cannot start before the method call arrives and a method cannot remain active after the method has returned. Do take note of this notation error.

yixuann02 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,342 @@
+the strings in the text file, instantiating an expenditure using the string received, and adding
+the expenditure into the array list.
+
+<p align="center">
+    <img src="team/images/initializeList.png">
+    <br/>
+    <i>Figure 6: Sequence diagram for the process of saveExpenditureList</i>
+</p>
+
```
Similarly, the return for the ExpenditureList, Expenditure etc. should not be before the end of activation bar, and also do take note of the notation errors mentioned above.

yixuann02 reviewed (commented):
Overall, very in-depth explanation of diagrams, good job! But do take note of the multiple notation errors in your diagrams. Well done.

jaredoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,342 @@
+
+Below shows the UML diagram representing the `command` package.
+<p align="center">
+    <img src="team/images/umlCommandClassDiagram.png" width="100%">
```
Good that .png format was used which ensured that the image remained clear even when zooming in to see the details. However, the details are quite small, so you might want to consider splitting it into different diagrams

jaredoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,342 @@
+</p>
+
+It must be noted that not all the existing parser commands are included in this sequence diagram for parsing, namely
+the mark, unmark and edit commands. This is because they have a similar sequence diagram as the functions parseAdd and 
+parseLendBorrow. The only difference is the condition, with the loop happening one, one and four time(s) respectively. 
```
The formatting for `parseAdd` and `parseLendBorrow` could be changed to standardize with the top portion.

jaredoong commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,342 @@
+`Expenditure`:
+- Fields: `date`, `amount`, `description`
+
+`AcademicExpenditure`**`: 
```
Additional `**`, might be a typo

jaredoong commented on docs/DeveloperGuide.md:
```diff
@@ -35,4 +348,212 @@
+- Navigate to the folder in command terminal and run the command `java -jar [filename].jar`
+- Alternatively, double click on the JAR file to run the app.
+#### Adding a record
+1. Adding an expenditure
```
Not sure why but the bullet point for this is not showing up correctly on the github pages, but you could possibly include a new line before this, which could fix the error.
![image](https://user-images.githubusercontent.com/88092096/229010975-75b0fba2-917c-43f8-ba49-d5d5757b2a68.png)

jaredoong commented on docs/DeveloperGuide.md:
```diff
@@ -35,4 +348,212 @@
+
+
+- Test case : `viewtype swimming`
+- Expected : Invalid message will be shown with the respective error message, in this case an
```
For each of the error message, it would be good to add an example to give the user a clearer idea on what the error message would look like

jaredoong reviewed (commented):
Overall clear developer guide with detailed and clear diagrams, just need to fix minor formatting error. Good job!

TzeLoong committed:
Merge remote-tracking branch 'upstream/master'

TzeLoong committed:
fix date bug

TzeLoong committed:
fix mark bug

TzeLoong committed:
update value bug

TzeLoong committed:
improve code quality for parseredit

TzeLoong committed:
Improve code quality for paseredit

TzeLoong committed:
Add PPP

dsicol committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T14-3/tp

itszhixuan committed:
Merge pull request #130 from TzeLoong/bugfixes

Bug fixes & Improve code quality & Add PPP

itszhixuan committed:
Added PPP

dsicol committed:
Add feature in TuitionExpenditure to keep track of repeat date

dsicol committed:
Add feature in AccommodationExpenditure to keep track of repeat date

dsicol committed:
Add methods to allow application to unmark paid field when a specific date passes

dsicol committed:
Refactor SortCommand and sorting methods in ExpenditureList

dsicol committed:
Fix bug when running the getNextRepeatDate method

dsicol committed:
Edit code style

dsicol committed:
Refactor code in txtfile

itszhixuan committed:
Fixed bug regarding inability to use check command for full list

itszhixuan committed:
Ensured check would not fail even if user inserts empty spaces after the command

itszhixuan committed:
Refactor case statement

itszhixuan committed:
Cleaned up code and added functionality of checking certain expenditure types

itszhixuan committed:
Addressed Gradle checks

dsicol committed:
Fix bugs and refactor code:

- Fix bug where next repeat date resets when reading from save file
- Add repeat date field to constructors of tuition and accommodation expenditures
- Add new repeat date field for save file
- Refactor code that processes the repeat date for tuition and accommodation expenditures

itszhixuan committed:
Merge pull request #132 from dsicol/enhancefeatures

Enhance features

TzeLoong committed:
Merge remote-tracking branch 'upstream/master'

dsicol committed:
Merge pull request #133 from itszhixuan/master

Improved code on `check` function

dsicol committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T14-3/tp

dsicol committed:
Add draft for dsicol PPP

Chick3nBoy committed:
Merge pull request #134 from dsicol/master

Add draft for PPP

itszhixuan committed:
Simplified `Parser` UML diagram to be readable

itszhixuan committed:
Added DG commands for `lend` and `borrow`, and indicating that the UML diagram for `lend` and `borrow` were already drawn under Parser component

itszhixuan committed:
Reformatted the DG display for manual testing with better formatting

itszhixuan committed:
Changed '/s' to '/p' for Edit, Add and Lend, Borrow

itszhixuan committed:
Replaced all 's/' to 'p/' in DG

dsicol committed:
Update dsicol PPP

Chick3nBoy committed:
Fix naming

Chick3nBoy committed:
Fix naming

itszhixuan committed:
Displaying `list` and `showrates`

itszhixuan committed:
delete and edit in DG

Chick3nBoy committed:
Add PPP and update UG

Chick3nBoy committed:
Merge pull request #135 from dsicol/master

Update PPP

itszhixuan committed:
Completed updates for DG instructions for manual testing

itszhixuan committed:
Replaced with 'p/' for tests

dsicol committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T14-3/tp

TzeLoong committed:
Merge pull request #137 from itszhixuan/master

Updated DG (Instructions for manual testing and Parser UML), replaced s/ to p/

TzeLoong committed:
Merge pull request #136 from Chick3nBoy/master

Update UG and PPP

TzeLoong committed:
Merge remote-tracking branch 'upstream/master'

dsicol committed:
Add tests for AccommodationExpenditure and TuitionExpenditure

dsicol committed:
Add newline for TuitionExpenditureTest.java

TzeLoong committed:
Fix bugs and update check budget

TzeLoong committed:
Fix code style violations

dsicol committed:
Fix bugs for repeat date and write BVA tests for repeat date

dsicol committed:
Edit method name in tests

dsicol committed:
Remove print lines in TuitionExpenditure.java and Accommodation.java

dsicol committed:
Edit test code to explicitly use a date to simulate the current date

dsicol committed:
Merge pull request #139 from TzeLoong/master

Fix bugs and update check budget

TzeLoong committed:
Merge pull request #138 from dsicol/junitv2.1

JUnit testing for repeating Accommodation and Tuition Expenditures

Chick3nBoy committed:
Add exception for negative value and error message for other misc fail case.

Chick3nBoy committed:
Update documentation for UG and PPP

itszhixuan committed:
UG slashes in names not allowed and not restricting description

itszhixuan committed:
Adding DG names no slashes

itszhixuan committed:
Merge pull request #140 from Chick3nBoy/master

Corrupted txtfile reading exception

Chick3nBoy committed:
Update DG

itszhixuan committed:
Include `check` documentation under UG and updated DG

dsicol committed:
Merge pull request #141 from itszhixuan/master

DG UG indicated that names cannot have slashes, Small UG changes

dsicol committed:
Merge pull request #142 from Chick3nBoy/master

Update DG

TzeLoong committed:
Update help message

TzeLoong committed:
change naming

itszhixuan committed:
Catch error where amounts more than 2dp can be added

itszhixuan committed:
Implemented catch for more than 2dp for edit command

itszhixuan committed:
Gradle check changes

itszhixuan committed:
Merge pull request #143 from TzeLoong/ImproveUI

Update help message

dsicol committed:
Fix txtfile bugs such that it will delete only corrupted expenditures

dsicol committed:
Add equivalence partition and boundary value analyses for repeat date tests

dsicol committed:
Merge pull request #144 from itszhixuan/master

Catches input amount that has more than 2dp

itszhixuan committed:
Merge pull request #145 from dsicol/bugs

Fix comprehensive txtfile bugs and add thorough tests for repeat date

itszhixuan committed:
Removed magic number

itszhixuan committed:
Update check documentation in help page

itszhixuan committed:
Numbering was previously inaccurate, changed it to follow sequential order

itszhixuan committed:
Added documentation for list CURRENCY in help function

TzeLoong committed:
Merge pull request #146 from itszhixuan/master

Documentation changes to help function

TzeLoong committed:
Fix bugs

itszhixuan committed:
Replaced magic number in check

TzeLoong committed:
Fix code style violations

itszhixuan committed:
User stories

itszhixuan committed:
Convert info in Command Component into a table as suggested in Issues

itszhixuan committed:
Update command class

itszhixuan committed:
Add list and showrates in table

itszhixuan committed:
Remove glossary from DG

itszhixuan committed:
Merge pull request #147 from TzeLoong/bugfixes

Bugfixes

itszhixuan committed:
PPP typo

dsicol committed:
Add more explanations to DG

itszhixuan committed:
UG DG minor updates

dsicol committed:
Format DG

Chick3nBoy committed:
Merge pull request #149 from dsicol/developerguide

Add more explanations in DG

itszhixuan committed:
Merge branch 'master' into master

TzeLoong committed:
Merge pull request #148 from itszhixuan/master

Update DG user stories and command component

TzeLoong committed:
Fix bug regarding negative amount for bug

TzeLoong committed:
Fix code style violations

Chick3nBoy committed:
Add check for large amount

dsicol committed:
Merge pull request #150 from TzeLoong/bugfixes

Fix bug regarding negative amount for bug

Chick3nBoy committed:
Fix styling

Chick3nBoy committed:
Update PPP

TzeLoong committed:
Add date checker

TzeLoong committed:
Fix code style violations

dsicol committed:
Add repeat date update in EditCommand

TzeLoong committed:
Fix code style violations

TzeLoong committed:
Fix code style violations

TzeLoong committed:
Fix code style violations

dsicol committed:
Edit check feature where marking as paid will add to expenditure total

dsicol committed:
Merge pull request #151 from Chick3nBoy/master

Add check for large amount

dsicol committed:
Merge branch 'master' into bugfixes

dsicol committed:
Change check feature to be able to add marked expenditures

dsicol committed:
Resolve merge conflict

dsicol committed:
Fix bugs in txtfile regarding corrupted amounts and lend/borrow dates

dsicol committed:
Fix code style

dsicol committed:
Update DG to explain repeatDate and save file corruptions

TzeLoong committed:
Fix code style violations

TzeLoong committed:
Merge pull request #152 from TzeLoong/bugfixes

Add date checker

TzeLoong committed:
Merge pull request #153 from dsicol/bugs

Fix textfile bugs when corrupting amount and dates

TzeLoong committed:
Fix bugs

TzeLoong committed:
Update test

TzeLoong committed:
Fix test

TzeLoong committed:
Update test

TzeLoong committed:
Update test

TzeLoong committed:
Update test

TzeLoong committed:
Fix test

TzeLoong committed:
Update test

itszhixuan committed:
check command in DG, missing UML diagram

itszhixuan committed:
Merge branch 'master' of https://github.com/itszhixuan/tp

TzeLoong committed:
Fix test

TzeLoong committed:
Update test

Chick3nBoy committed:
Merge pull request #154 from TzeLoong/bugfixes

Fix bugs

itszhixuan committed:
`check` puml diagram

itszhixuan committed:
typo in command

itszhixuan committed:
small changes

itszhixuan committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T14-3/tp

itszhixuan committed:
Add `mark` `unmark` in DG, and removed output errors in DG

Chick3nBoy committed:
Update docs and help message

Chick3nBoy committed:
Merge pull request #155 from itszhixuan/master

UML diagram for `check` command in DG

itszhixuan committed:
Merge branch 'master' into master

itszhixuan committed:
Merge pull request #156 from Chick3nBoy/master

Update docs and help message

itszhixuan committed:
Mark unmark in manual testing DG

itszhixuan committed:
UG extra characters removed

itszhixuan committed:
Added already marked in DG test case

itszhixuan committed:
Changed `mark` documentation in UG and added new `check` commands under command summary

dsicol committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T14-3/tp

dsicol committed:
Merge branch 'master' into developerguide

itszhixuan committed:
Update parser message in DG

itszhixuan committed:
Missing field in table

Chick3nBoy committed:
Merge pull request #157 from itszhixuan/master

DG UG changes to fit v2.1

itszhixuan committed:
Add limits to year in UG

Chick3nBoy committed:
Add page breaks to UG

Chick3nBoy committed:
Merge pull request #158 from itszhixuan/master

Add limits to year in UG

itszhixuan committed:
Merge pull request #159 from Chick3nBoy/master

Add page breaks to UG

Chick3nBoy committed:
Add page break to DG

itszhixuan committed:
Major typo in UG

itszhixuan committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T14-3/tp

itszhixuan committed:
Exceptions for ParseLendBorrow

itszhixuan committed:
Spliting of lines tha overshoot in pdf form in ug

Chick3nBoy committed:
Merge pull request #160 from itszhixuan/master

Exceptions for ParseLendBorrow, and bug in UG

itszhixuan committed:
Merge pull request #161 from Chick3nBoy/master

Add page break to DG

dsicol committed:
Update DG with repeatDate diagrams and command classes

Chick3nBoy committed:
Merge pull request #162 from dsicol/developerguide

Developer guide updates

TzeLoong committed:
Update error message

TzeLoong committed:
Update Description

dsicol committed:
Update PPP

dsicol committed:
Merge pull request #163 from TzeLoong/UpdateUserguide

Update error message

TzeLoong committed:
Update parseLendBorrow

TzeLoong committed:
Update help message

itszhixuan committed:
Merge pull request #165 from TzeLoong/UpdateUserguide

Update userguide

itszhixuan committed:
Merge pull request #164 from dsicol/developerguide

Update PPP

dsicol committed:
Format DG

TzeLoong committed:
Fix editCommand bug

dsicol committed:
Edit style for DG

TzeLoong committed:
Update userguide

itszhixuan committed:
Merge pull request #166 from dsicol/developerguide

Format DG

TzeLoong committed:
Update userguide

itszhixuan committed:
Check on DG and UG

itszhixuan committed:
Justified why some outputs are across 2 lines

itszhixuan committed:
update ppp

itszhixuan committed:
missing INDEX in edit command in UG

dsicol committed:
Merge pull request #168 from itszhixuan/master

Formatting changes on UG and DG after check

dsicol committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T14-3/tp

dsicol committed:
Merge branch 'master' into developerguide

TzeLoong committed:
Update editCommand

TzeLoong committed:
Fix bugs

dsicol committed:
Update preface for UG and DG

itszhixuan committed:
Updated README

TzeLoong committed:
Fix textfile date bug

Chick3nBoy committed:
Merge pull request #169 from itszhixuan/master

Updated README

TzeLoong committed:
Update developerGuide

TzeLoong committed:
Update DeveloperGuide.md

Chick3nBoy committed:
Merge branch 'master' into UpdateUserguide

Chick3nBoy committed:
Merge pull request #167 from TzeLoong/UpdateUserguide

Update userguide

dsicol committed:
Merge branch 'developerguide'

dsicol committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T14-3/tp

itszhixuan committed:
Merge pull request #170 from dsicol/developerguide

Developer guide edits

Chick3nBoy committed:
Add page break

Chick3nBoy committed:
Add page break

itszhixuan committed:
page break pt 1

itszhixuan committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T14-3/tp

itszhixuan committed:
page break pt 2

Chick3nBoy committed:
Update UG

TzeLoong committed:
Update Test

itszhixuan committed:
Remove month

Chick3nBoy committed:
Update UG

dsicol committed:
Merge pull request #171 from itszhixuan/master

DG updates

TzeLoong committed:
Fix code style violations

Chick3nBoy committed:
Update UG

TzeLoong committed:
revert changes

Chick3nBoy committed:
Update UG

TzeLoong committed:
Update userguide

dsicol committed:
Reformat PPP

Chick3nBoy committed:
Update UG

dsicol committed:
Merge branch 'master' into master

dsicol committed:
Merge pull request #172 from TzeLoong/UpdateUserguide

Update Test

Chick3nBoy committed:
Update UG

TzeLoong committed:
Merge pull request #173 from dsicol/master

Reformat PPP

itszhixuan committed:
Reformat DG

dsicol committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T14-3/tp

dsicol committed:
Merge branch 'master' into developerguide

Chick3nBoy committed:
Add page break

dsicol committed:
Add page breaks

dsicol committed:
Merge pull request #174 from itszhixuan/master

Reformat DG

itszhixuan committed:
Merge pull request #175 from dsicol/developerguide

Add page breaks for PPP

Chick3nBoy committed:
Merge branch 'master' into master

dsicol committed:
Improve format of PPP

dsicol committed:
Merge pull request #176 from Chick3nBoy/master

Finalized UG

Chick3nBoy committed:
Merge pull request #177 from dsicol/developerguide

Improve format of PPP

Chick3nBoy committed:
Update UG

Chick3nBoy committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-T14-3/tp

itszhixuan committed:
new update formating

dsicol committed:
Remove numbering from PPP

itszhixuan committed:
Merge pull request #179 from dsicol/master

Remove numbering from PPP

dsicol committed:
Merge pull request #178 from itszhixuan/master

new update formating

Chick3nBoy committed:
Update UG

itszhixuan committed:
edit

itszhixuan committed:
page break

TzeLoong committed:
Merge pull request #180 from itszhixuan/master

edit

itszhixuan committed:
<br>

dsicol committed:
Fix style in PPP and create JAR

itszhixuan committed:
double page break removed

dsicol committed:
Merge pull request #181 from Chick3nBoy/master

Final finalized DG

dsicol committed:
Merge pull request #182 from itszhixuan/master

<br> Should not have errors anymore

itszhixuan committed:
more edits

dsicol committed:
Merge pull request #184 from itszhixuan/master

more formatting DG

dsicol committed:
Merge pull request #183 from dsicol/master

Fix style in PPP and create JAR

dsicol committed:
Reformat PPP

itszhixuan committed:
Merge pull request #185 from dsicol/master

Reformat PPP
